src/server/acl/acl_family.cc

```diff
@@ -27,9 +27,9 @@ static std::string AclToString(uint32_t acl_category) {
   const std::string prefix = "+@";
   const std::string postfix = " ";
 
-  for (uint32_t step = 0, cat = 0; cat != JSON; cat = 1ULL << ++step) {
+  for (const auto& [cat_name, cat] : CATEGORY_INDEX_TABLE) {
```

Comment:
The issue here is that `CATEGORY_INDEX_TABLE` is a hash table, and the order of traversal is undefined so it's really hard to test this on pytests. 

A better fix would be to move the bits in the bitfield to make them contiguous. If redis adds more categories in the future we can move them accordingly, it won't affect the UX

Reply:
Fair. I opted for just putting dummy values in the `REVERSE_CATEGORY_INDEX_TABLE` list and simplified the loop.